packages/tools/devtools/devtools-core/src/ContainerRuntimeDevtools.ts

```diff
@@ -155,10 +155,17 @@ export class DecomposedContainerForContainerRuntime
 	}
 
 	public get connectionState(): ConnectionState {
+		// Normal connection state logic - readonly state is handled separately via readOnlyInfo
 		return this.runtime.connected ? ConnectionState.Connected : ConnectionState.Disconnected;
 	}
 
 	public get closed(): boolean {
 		return this._disposed; // IContainerRuntime doesn't have a "closed" state - only "disconnected" (reconnectable) and "disposed" (permanent)
 	}
+
+	public get readOnlyInfo(): { readonly readonly?: boolean } {
+		// IContainerRuntime doesn't expose readonly in its interface, but the implementation has isReadOnly()
+		const runtimeWithReadOnly = this.runtime as { isReadOnly?: () => boolean };
+		return { readonly: runtimeWithReadOnly.isReadOnly?.() };
```

Comment:
This will be brittle to changes in `ContainerRuntime`. I would recommend doing the following instead, so we detect in the future at build time if `ContainerRuntime` is ever updated in a way that would break this code:

```suggestion
		assert(this.runtime instanceof ContainerRuntime);
		return { readonly: this.runtime.isReadOnly() };
```

Reply:
We could further improve this (since we now depend on the implementation rather than the interface) by storing the runtime as `ContainerRuntime` instead of `IContainerRuntime` and asserting this in the `ContainerRuntimeDevtools` constructor. This way we "fail fast", and the user is made aware of type incompatibility at construction, rather than when the tool goes to check this property.